pkg/router/template/configmanager/haproxy/manager.go

```diff
@@ -915,8 +928,18 @@ func (entry *routeBackendEntry) BuildMapAssociations(route *routeapi.Route) {
 	}
 }
 
+// validateBlueprintRoute runs extended validation on a blueprint route.
+func validateBlueprintRoute(route *routeapi.Route) error {
```

Comment:
I think you can actually delete this entire function and replace all usages with:

```go
if err := validation.ExtendedValidateRoute(newRoute).ToAggregate(); err != nil {
	glog.Errorf("Skipping blueprint route %s/%s due to invalid configuration: %s", route.Namespace, route.Name, err)
}
```